app/models/infrastructure_facades/plgrid/grid_schedulers/qsub.rb

```diff
@@ -163,6 +163,15 @@ def job_pbs_file(sm_uuid)
       "scalarm_job_#{sm_uuid}.pbs"
     end
 
+    ##
+    # Returns list of hashes representing distinct types of infrastructure
+    # Subinfrastructures are distinguished by:
+    #  * grant ids
+    def get_subinfrastructures(user_id)
+      PlGridFacade.retrieve_grants(GridCredentials.find_by_user_id(user_id)).flat_map do |grant_id|
```

Comment:
Tego wcześniej nie zauważyłem - GridCredentials są usuwane jeśli są już niepotrzebne, tzn. użytkownik się wylogował i działa onsite_monitoring usuwane jest proxy. ``get_subinfrastructures`` jest wołane gdzieś w tle, czy tylko podczas wyświetlania GUI z możliwymi infrastrukturami? Dobrze by było to przetestować w jakichś dziwnych warunkach (np. użytkownik nie zalogował się za pomocą OpenID i retrieve_grants nie zadziała, bo nie ma GridCredentials).